### PR TITLE
Add the common test e2e files

### PR DESCRIPTION
**Pull request by srinivashegde86**
Part of #26. 

Start moving the common e2e test logic to pkg.

The existing files will be deleted and the references will be updated in another PR

### CODE REVIEW

File: test/e2e_flags.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	flag.StringVar(&f.Namespace, "namespace", "",
+		"Provide the namespace you would like to use for these tests.")
+
+	flag.BoolVar(&f.LogVerbose, "logverbose", false,
+		"Set this flag to true if you would like to see verbose logging.")
+
+	flag.BoolVar(&f.EmitMetrics, "emitmetrics", false,
+		"Set this flag to true if you would like tests to emit metrics, e.g. latency of resources being realized in the system.")
+
+	flag.Parse()
```
Comment (mattmoor) [inline test/e2e_flags.go]: This might be a good use of a FlagSet.

Comment (srinivashegde86) [reply test/e2e_flags.go]: https://github.com/knative/serving/issues/1845

We have an owners file for test/*. This approval is for other files in root dir. Thanks.

**Pull request merged**